### PR TITLE
use big-bang for IO and refactor with a more functional style

### PR DESCRIPTION
**Pull request by chikeabuah**

### CODE REVIEW

File: main.rkt [hunk 1]
```diff
@@ -7,9 +7,12 @@
    racket/syntax
    racket/match)
   graph
+  2htdp/universe
```
Comment (jeapostrophe) [inline main.rkt]: You should figure out a way to parameterize it so it doesn't have to be part of the main library.

File: main.rkt [hunk 2]
```diff
@@ -174,13 +170,53 @@
...
+       (- (current-seconds) (start-time))))
+    (current-events (hash-union (current-events) events-so-far #:combine (λ (old new) new)))
+    ;;then step
+    (displayln "executing recess graph...")
+    (step-world)
+    ;; reset pending events and produce output
+    ;; need to reset sink events too
+    (struct-copy big-bang-recess-world w
+                 [pending-events (make-immutable-hasheq)]
+                 [last-output (list)])))
```
Comment (jeapostrophe) [inline main.rkt]: This is very pretty

Comment (chikeabuah) [reply main.rkt]: thank you! :)

File: main.rkt (hunk 1 above)
Comment (chikeabuah) [reply main.rkt]: @jeapostrophe do you mean something like: passing 2htdp/universe into recess as an argument from the module where it's being used?

Comment (jeapostrophe) [reply main.rkt]: No, I mean figuring out a nice interface like the "init" / "step" functions we talked about before then just writing this code in another module so you can be sure that it is not baked into recess

Comment (chikeabuah) [reply main.rkt]: okay! thank you!

File: big-bang-step.rkt [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+(require 2htdp/universe 2htdp/image racket/hash)
+
+(provide
+ (all-defined-out))
+
+;; adapt recess to use big bang
+(struct big-bang-recess-world (pending-events recess-state last-output) #:transparent)
+
+(define (big-bang-step start-time stop-func current-events step-world)
```
Comment (chikeabuah) [inline big-bang-step.rkt]: @jeapostrophe i hope this is similar to what you wanted

**Pull request merged**